pkg/collector/corechecks/cluster/orchestrator/processors/k8s/pod_tag_provider/provider.go

```diff
@@ -0,0 +1,37 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build orchestrator
+
+// Package podtagprovider implements a provider that supplies pod tags for cluster checks
+package podtagprovider
+
+import (
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	taggertypes "github.com/DataDog/datadog-agent/comp/core/tagger/types"
+	workloadmeta "github.com/DataDog/datadog-agent/comp/core/workloadmeta/def"
+	pkgconfigsetup "github.com/DataDog/datadog-agent/pkg/config/setup"
+
+	"github.com/DataDog/datadog-agent/pkg/util/flavor"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+// PodTagProvider can be used to extract pod tags
+type PodTagProvider interface {
+	GetTags(*corev1.Pod, taggertypes.TagCardinality) ([]string, error)
+}
+
+// NewPodTagProvider returns a new PodTagProvider
+// In case of CLC runner, the provider will calculate tags based on the pod resource on demand each time
+// In case of Node agent or Cluster agent, the provider will get pod tags by querying the tagger
+func NewPodTagProvider(cfg config.Component, store workloadmeta.Component) PodTagProvider {
+	if flavor.GetFlavor() != flavor.ClusterAgent && pkgconfigsetup.IsCLCRunner(pkgconfigsetup.Datadog()) {
+		// Running in a CLC Runner
```

Comment:
Isn't the first condition of the `if` redundant if this is only about CLC?

Reply:
Actually here I want to distinguish between two cases:
- running the check in a node agent
- running the check in a cluster check runner

Checking `isCLCRunner()` alone is not enough to make the distinction.

But you are still correct that the first condition is redundant. However, I think the statement should be changed to this:

```
if flavor.GetFlavor() != flavor.ClusterAgent && !pkgconfigsetup.IsCLCRunner(pkgconfigsetup.Datadog()) {
// this means we are in the orchestrator check version that runs in the node agent
return newNodePodTagProvider()
}

return newCLCTagProvider(cfg, store)
```

WDYT?